.travis.yml

```diff
@@ -85,14 +45,67 @@ before_script:
 script:
   - ./travis_configure.sh run_tests
 
-## Deploy debian packages to bintray
-##
-## deb package built on ubuntu xenial with gcc
-deploy:
-    provider: script
-    script: beaver bintray upload -D $DIST_VERSION -N pkg/deb/*.deb
-    skip_cleanup: true
-    on:
-        tags: true # must be a git tag
-        repo: sociomantic-tsunami/libdrizzle-redux # must be upstream
-        condition: $TRAVIS_OS_NAME = 'linux' && $DIST_NAME = 'ubuntu' && $DIST_VERSION = 'xenial' && $CC = 'gcc' && $MAKE_TARGET =~ 'deb' && $DEPLOY_PACKAGE = 'true'
+## Defines the order of the build stages.
+#
+#  The deployment stage is only done is building semver git tag
+stages:
+  - test
+  - deploy
+
+## Defines the jobs to run in each stage
+#
+#  Travis combines the variables defined in env.global with the variables
+#  defined for each job
+#
+#  The resulting build matrix is as follows:
+#
+#  | os     | dist   | compiler  | make targets | deploy |
+#  |--------|--------|-----------|--------------|--------|
+#  | linux  | ubuntu | clang     | check        | no     |
+#  | linux  | ubuntu | gcc       | check deb    | yes    |
+#  | linux  | centos | gcc       | check rpm    | yes    |
+#  | osx    | sierra | clang¹    | check        | no     |
+#
+#  Linux based builds are run in docker containers using docker-compose.
+#
+#  OSX based builds are done by a third-party provider
+#
+#  ¹ The compiler reported is the Apple LLVM version
+jobs:
+  templates:
+    # Global config for builds on linux
+    - &linux-build
+      services:
+        - docker
+    # Global config for builds on osx
+    - &osx-build
+      os: osx
+      compiler: clang
+    # Global config for deployment stage
+    - &deploy
+      os: linux
```

Comment:
Sorry if this was there long ago, but I just see it.

These templates look really funny, it seems what they include looks pretty arbitrary and also the OSX and deploy template are only used once. I would  only move stuff to a template when you really need to reuse it, otherwise it just makes following this file more complicated.

You can leave it for a future PR though.

Reply:
> These templates look really funny, it seems what they include looks pretty arbitrary 

You the `services.docker`? The reason for that is that docker isn't needed in the deploy stage

Reply:
>  also the OSX and deploy template are only used once

OSX: I know, but basically just to prepare it for building with different versions of XCode
Deploy: There is another PR for building coverage (`test-stage`) and uploading (`deploy-stage`) it to `coveralls.io`.